### PR TITLE
created new folders and moved sum function to the utils module

### PR DESCRIPTION
**Pull request by gabidoye**

### CODE REVIEW

File: src/gpa.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+         {"title": "English", "unit": 2, "code": "ENG101", "score": 60, "term": "Fall", "session":'2020'},
+         {"title": "Chemistry I", "unit": 4, "code": "CHE101", "score": 70, "term": "Fall", "session":'2021'},
+         {"title": "Maths", "unit": 3, "code": "MTH101", "score": 80, "term": "Spring", "session":'2020'},
+         {"title": "Chemistry II", "unit": 4, "code": "CHE102", "score": 91, "term": "Spring", "session":'2021'}, 
+         {"title": "History", "unit": 2, "code": "HIS102", "score": 40, "term": "Spring", "session":'2020'}
+      ]
+   }
+
+   results=compute_results(student_records)
+   print(results)
```
Comment (eibidun) [inline src/gpa.py]: These should go in `entry.py`

File: src/utils.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+def calculate_sum(lst):
+   """
+   This function sums nested list.
+   parameters
+   -----------
+   list 
+   return
+   -------
+   array of values
```
Comment (eibidun) [inline src/utils.py]:
```suggestion
   
   Parameters
   -----------
   lst: List
       A list of values to be summed.
       
   Returns
   --------
   total: int
         sum of all input values
```

Please apply the similar changes as suggested below to all other functions.

File: src/utils.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+   """
+      
+   return list(map(sum, zip(*lst)))
```
Comment (eibidun) [inline src/utils.py]:
```suggestion
   total =  list(map(sum, zip(*lst)))
   return total
```

**Review by eibidun — CHANGES_REQUESTED** [review]
Looks good. Fix the suggested changes and include a proof that it works in the PR description.

File: src/gpa.py (hunk 1 above)
Comment (gabidoye) [reply src/gpa.py]: Fixed.
image

Comment (gabidoye) [reply src/gpa.py]: prove attached.

File: src/utils.py (hunk 2 above)
Comment (gabidoye) [reply src/utils.py]: Fixed

File: entry.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import sys
+    # caution: path[0] is reserved for script path (or '' in REPL)
+sys.path.insert(1, '/Users/gabidoye/Documents/data_Engineering/students_cgpa_python/src')
```
Comment (manuelibidun) [inline entry.py]: Why do you need this?

Comment (gabidoye) [reply entry.py]: using it to read file from another folder.

**Review by manuelibidun — CHANGES_REQUESTED** [review]
Looks good! 
Please do the following: 

- add a `__init__.py` to `src/` it might fix my inline comment
- add same to all folders proposed in our chat last time 
   - this will make them show up in github (by default, github will not add empty directories)
-  if you're curious about why....see:
    - https://www.tutorialspoint.com/What-is-init-py-in-Python 
    - https://careerkarma.com/blog/what-is-init-py/
    - https://stackoverflow.com/questions/448271/what-is-init-py-for

**Comment by gabidoye** [comment]
The __init__.py Fixed the module import issues.  Have also added to all folder as suggested.

File: data/dummy_data.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+# for _ in range(100):
+#     fake_data["first_name"].append( fake.first_name() )
+#     fake_data["last_name"].append( fake.last_name() )
+#     fake_data["occupation"].append( fake.job() )
+#     fake_data["dob"].append( fake.date_of_birth() )
+#     fake_data["country"].append( fake.country() )
+
+# df_fake_data = pd.DataFrame(fake_data)
+
+# df_fake_data.shape
```
Comment (manuelibidun) [inline data/dummy_data.py]: Remove commented code in the final commits accompanying a PR. If you need to refer to them later, you can refer to specific commits.

**Review by manuelibidun — DISMISSED** [review]
For future contributions: 

- don't commit `.DS_Store` files to version control; use [gitignore](https://github.com/github/gitignore) for that
- don't commit commented codes to version control...

**Comment by gabidoye** [comment]
suggestions implemented

**Review by manuelibidun — APPROVED** [review]

**Pull request merged**